src/discovery/factory.rs

```diff
@@ -33,9 +35,8 @@ impl DiscoverableFactory {
 // Returns a vec of empty factories that match one of the Factory interfaces specified by each DiscoverableFactory
 pub async fn discover_factories<M: Middleware>(
     factories: Vec<DiscoverableFactory>,
-    number_of_amms_threshold: u64,
     middleware: Arc<M>,
-    step: u64,
+    options: FactoriesDiscoveryOptions,
```

Comment:
Same thing here, can you make this optional and then use a default implementation if this is None?

Reply:
fixed